examples/marquee.lisp

```diff
@@ -0,0 +1,88 @@
+(defpackage #:charms-marquee
+  (:use #:cl)
+  (:export #:main))
+(in-package :charms-marquee)
+
+(defclass marquee ()
+  ((position :initarg :position :accessor marquee-position)
+   (size :initarg :size :accessor marquee-size)
+   (text :initarg :text :accessor marquee-text)
+   (viewport :initarg :viewport :reader marquee-viewport)
+   (padding-char :initarg :padding-char :initform #\Space :accessor marquee-padding-char)
+   (buffer :initform "" :accessor marquee-buffer))
+  (:documentation "A marquee is like a scrolling text thing. A string of characters moves through
+a viewport. The viewport constrains the text so you only see a little bit at a time."))
+
+;; (defmethod marquee-viewport :before ((marquee marquee))
+;;   "Before getting the MARQUEE viewport (the text that is shown at a given time), the buffer
+;; needs to be padded in case the text doesn't fill the MARQUEE size"
+;;  ( )
+
+(defmethod marquee-viewport ((marquee marquee))
```

Comment:
for this and below, I personally advise just making everything functions with DEFUN. It's not required, but when I see DEFMETHOD:

- I generally expect there will be a matching DEFGENERIC elsewhere establishing an abstract protocol, or
- I expect these methods to be reasonable targets for implementing additional functionality for other classes.

I don't think either is the case here, so I think DEFGENERIC signals a red herring to the reader.